distributed/tests/test_worker.py

```diff
@@ -55,6 +56,109 @@ def test_health():
         assert 'network-send' in d
 
 
+def test_worker_bad_args(loop, capsys):
+    @gen.coroutine
+    def f(c, a, b):
+        aa = rpc(ip=a.ip, port=a.port)
+        bb = rpc(ip=b.ip, port=b.port)
+
+        class NoReprObj(object):
+            def __str__(self):
+                raise ValueError("I have no str representation.")
+            def __repr__(self):
+                raise ValueError("I have no repr representation.")
+
+        response = yield aa.compute(key='x',
+                                    function=dumps(NoReprObj),
+                                    args=dumps(()),
+                                    who_has={})
+        assert not a.active
+        assert response['status'] == 'OK'
+        assert a.data['x']
+        assert c.who_has['x'] == {a.address}
+        assert isinstance(response['compute_start'], float)
+        assert isinstance(response['compute_stop'], float)
+        assert isinstance(response['thread'], Integral)
+
+        def bad_func(*args, **kwargs):
+            1 / 0
+
+        class MockLoggingHandler(logging.Handler):
+            """Mock logging handler to check for expected logs."""
+
+            def __init__(self, *args, **kwargs):
+                self.reset()
+                logging.Handler.__init__(self, *args, **kwargs)
+
+            def emit(self, record):
+                self.messages[record.levelname.lower()].append(record.getMessage())
+
+            def reset(self):
+                self.messages = {
+                    'debug': [],
+                    'info': [],
+                    'warning': [],
+                    'error': [],
+                    'critical': [],
+                }
+
+        hdlr = MockLoggingHandler()
+        logger.addHandler(hdlr)
+        response = yield bb.compute(key='y',
+                                    function=dumps(bad_func),
+                                    args=dumps(['x']),
+                                    kwargs=dumps({'k': 'x'}),
+                                    who_has={'x': [a.address]})
+        assert not b.active
+        assert response['status'] == 'error'
+        assert isinstance(loads(response['exception']), ZeroDivisionError)
+        if sys.version_info[0] >= 3:
+            assert any('1 / 0' in line
+                      for line in pluck(3, traceback.extract_tb(
+                          loads(response['traceback'])))
+                      if line)
+        assert hdlr.messages['warning'][0] == " Compute Failed\n" \
+            "Function: bad_func\n" \
+            "args:     (< could not convert arg to str >)\n" \
+            "kwargs:   {'k': < could not convert arg to str >}\n"
+
```

Comment:
Ah, sorry, I thought it was odd that you wrote above, but these were the lines that were shown next to your comment.
The lines below are to check that none of the workers died as a result of processing the sad object - so they're still accepting jobs and processing them normally. Perhaps a comment is in place. I will add.